controllers/managedmcg_controller.go

```diff
@@ -143,12 +140,14 @@ func (r *ManagedMCGReconciler) Reconcile(_ context.Context, req ctrl.Request) (c
 	if r.managedMCG.UID != "" {
 		statusErr = r.Client.Status().Update(r.ctx, r.managedMCG)
 	}
+
 	// Reconcile errors have priority to status update errors
-	if err != nil {
-		return ctrl.Result{}, err
-	} else if statusErr != nil {
-		return ctrl.Result{}, statusErr
-	} else {
+	switch {
```

Comment:
By replacing elseif with switch what we are trying to achieve here?

Reply:
This change is suggested by the `gocognit` linter. PTAL https://github.com/uudashr/gocognit#understanding-the-complexity.